src/utils/sendPaginatedList.js

```diff
@@ -18,16 +18,21 @@ async function sendPaginatedList(res, Model, filter, pageNo, limit, select, mapp
     if (pageCount < pageNo) {
       return sendFailedResponse(res, 'It is not possible to split into so many elements');
     }
-    const filteredItemList = await Model.find(filter).skip((pageNo - 1) * limit).limit(limit).select(select);
-    const itemList = filteredItemList.map((elem) => {
-      mapKeys.forEach((key) => {
-        if (elem[key]) {
-          elem[mapping[key]] = elem[key];
-          delete elem[key];
-        }
+    if (mapping) {
+      const mapKeys = Object.keys(mapping);
+      const filteredItemList = await Model.find(filter).skip((pageNo - 1) * limit).limit(limit).select(select);
+      itemList = filteredItemList.map((elem) => {
+        mapKeys.forEach((key) => {
+          if (elem[key]) {
+            elem[mapping[key]] = elem[key];
+            delete elem[key];
+          }
+        });
+        return elem;
       });
-      return elem;
-    });
+    } else {
+      itemList = await Model.find(filter).skip((pageNo - 1) * limit).limit(limit);
+    }
     return sendSuccessResponse(res, message, {
```

Comment:
Lets change this `if else` block. So after checking if pageNo is valid(lines 18-20) lets have.

```
// filter, pageNo, and limit are required, we have those keys for sure
const fetchPromise = Model.find(filter).skip((pageNo - 1) * limit).limit(limit);
if (select) { // Adding select to our findPromise, if it exists
  fetchPromise.select(select);
}
const itemList = await fetchPromise();
if (mapping) {
   // Do the mapping and return and send response with the mapped list
}
return itemList; // Return list without mapping
```